monai/transforms/io/array.py

```
@@ -206,13 +210,24 @@ def __init__(
                 if not has_built_in:
                     the_reader = locate(f"{_r}")  # search dotted path
                 if the_reader is None:
-                    the_reader = look_up_option(_r.lower(), SUPPORTED_READERS)
+                    try:
+                        the_reader = look_up_option(_r.lower(), SUPPORTED_READERS)
+                    except ValueError:
+                        # If the reader name is not recognized at all, raise OptionalImportError
+                        msg = f"Cannot find reader '{_r}'. It may not be installed or recognized."
+                        if self.raise_on_missing_reader:
+                            raise OptionalImportError(msg)
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)